From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Ledbetter <timledbetter@gmail.com>
Date: Sat, 5 Aug 2023 09:57:15 +0100
Subject: [PATCH] Update the makefile for serenity

This lets the build system choose the compiler. Also, `-std=c17` is
required as newer version are not compatible with custom definition of
bool, true and false. Finally, it seems that newer versions of GCC
report an error for incompatible-pointer-types, so turn this into a
warning.
---
 doomgeneric/Makefile.sdl | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/doomgeneric/Makefile.sdl b/doomgeneric/Makefile.sdl
index 38402da57ac889901b7080341a842a2acbdfe90b..ceb0349d231816728581696142f0e9e484fcd720 100644
--- a/doomgeneric/Makefile.sdl
+++ b/doomgeneric/Makefile.sdl
@@ -12,18 +12,17 @@ else
 endif
 
 
-SDL_CFLAGS = `sdl2-config --cflags`
-SDL_LIBS = `sdl2-config --cflags --libs` -lSDL2_mixer
+SDL_CFLAGS = -I$(SERENITY_INSTALL_ROOT)/usr/local/include/SDL2 -D_REENTRANT
+SDL_LIBS = -L$(SERENITY_INSTALL_ROOT)/usr/local/lib -lSDL2 -lSDL2_mixer
 
 
-CC=clang  # gcc or g++
-CFLAGS+=-DFEATURE_SOUND $(SDL_CFLAGS)
+CFLAGS+=-DFEATURE_SOUND $(SDL_CFLAGS) -std=c17 -Wno-error=incompatible-pointer-types
 LDFLAGS+=
 LIBS+=-lm -lc $(SDL_LIBS)
 
 # subdirectory for objects
 OBJDIR=build
-OUTPUT=doomgeneric
+OUTPUT=doom
 
 SRC_DOOM = dummy.o am_map.o doomdef.o doomstat.o dstrings.o d_event.o d_items.o d_iwad.o d_loop.o d_main.o d_mode.o d_net.o f_finale.o f_wipe.o g_game.o hu_lib.o hu_stuff.o info.o i_cdmus.o i_endoom.o i_joystick.o i_scale.o i_sound.o i_system.o i_timer.o memio.o m_argv.o m_bbox.o m_cheat.o m_config.o m_controls.o m_fixed.o m_menu.o m_misc.o m_random.o p_ceilng.o p_doors.o p_enemy.o p_floor.o p_inter.o p_lights.o p_map.o p_maputl.o p_mobj.o p_plats.o p_pspr.o p_saveg.o p_setup.o p_sight.o p_spec.o p_switch.o p_telept.o p_tick.o p_user.o r_bsp.o r_data.o r_draw.o r_main.o r_plane.o r_segs.o r_sky.o r_things.o sha1.o sounds.o statdump.o st_lib.o st_stuff.o s_sound.o tables.o v_video.o wi_stuff.o w_checksum.o w_file.o w_main.o w_wad.o z_zone.o w_file_stdc.o i_input.o i_video.o doomgeneric.o doomgeneric_sdl.o mus2mid.o i_sdlmusic.o i_sdlsound.o
 OBJS += $(addprefix $(OBJDIR)/, $(SRC_DOOM))
@@ -53,3 +52,6 @@ $(OBJDIR)/%.o:	%.c
 print:
 	@echo OBJS: $(OBJS)
 
+install:
+	mkdir -p $(DESTDIR)/usr/local/bin
+	cp $(OUTPUT) $(DESTDIR)/usr/local/bin
